js/dom.js

```diff
@@ -357,7 +357,8 @@ function focusAccessibility() {
   const isOutlineAllowed = ({localName, type}) =>
     !focusAccessibility.ELEMENTS.includes(localName) ||
     // allow outline on text/search inputs in addition to textareas
-    localName === 'input' && (type === 'text' || type === 'search');
+    localName === 'input' &&
+    !/^(button|checkbox|color|file|image|radio|submit|range)$/.test(type);
```

Comment:
I would use a positive check for `/^(text|search|number)$/` because it's shorter

Reply:
Well, it's future proof, plus some text inputs don't have type attributes.

Reply:
`type` is never empty inside JS so it'll be `text`. As for the future, I could argue a fancy control is more likely to be added whereas the classic ones for text/search/number are more likely to keep going for ages.

Reply:
>`type` is never empty inside JS so it'll be `text`.

News to me. Some weren't working, but it must've just been the data attribute on a parent. Positive check should be fine then.

Reply:
Yeah by "JS" I mean direct access to the element's property like we do here, not via getAttribute or dataset and such.